autorest/validations/validations.go

```diff
@@ -0,0 +1,415 @@
+/*
+Package validations provides methods for validating parameter value using reflection.
+*/
+package validations
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+// Constraint stores constraint name, target field name
+// Rule and chain validations.
+type Constraint struct {
+
+	// Target field name for validation.
+	Target string
+
+	// Constraint name e.g. Minlength, Maxlength, Pattern, etc.
+	Name int
+
+	// Rule for constraint e.g. greater than 10, less than 5 etc.
+	Rule interface{}
+
+	// Chain Validations for struct type
+	Chain []Constraint
+}
+
+// Validation stores parameter-wise validation.
+type Validation struct {
+	TargetValue interface{}
+	Constraints []Constraint
+}
+
+// Constraints
+const (
+	InclusiveMaximum = iota + 1
```

Comment:
not sure where the numbers are coming from, but why does this start from iota+1?